build/parameters.go

```diff
@@ -82,3 +82,5 @@ const Eip155ChainId = buildconstants.Eip155ChainId // Deprecated: Use buildconst
 var WhitelistedBlock = buildconstants.WhitelistedBlock // Deprecated: Use buildconstants.WhitelistedBlock instead
 
 const Finality = policy.ChainFinality // Deprecated: Use policy.ChainFinality instead
+
+const SafeHeightDistance = 200
```

Comment:
If we are going to let the concept of safe be exposed from more areas than just the eth APIs, I think we should make it consistent.  Basically, is there a reason for ETH APIs to use one value and a different value here?

Ideally can we just use https://github.com/filecoin-project/lotus/blob/811d25c3e29bbda4d188d78f6b41c5056c378ccf/chain/types/ethtypes/eth_types.go#L46 or have that use this build parameter?  And if it's defined both places, cross link it (i.e., if you change this value here please also change over there).

Reply:
Happy to change it, but if exchanges don't use the distance of 30 then I have left wondering how useful this tag would be in the context we discussed with Zondax.

Of course exchanges can use a different build param and pick whatever number they want. 

Another key question is: Despite what's being picked for Eth APIs, what do we want the official `safe` distance to be for Filecoin v2 APIs?

Having reviewed FRC-0089, at least we a have math behind the distance 30 which is more than I can say about the distance 200.

Reply:
I have reached out to Zondax to ask about 200 vs 30 for filecoin tipsets.

Reply:
@masih this should be in `build/buildconstants/params_shared_vals.go`, don't even leave an alias for it, the rest of these are deprecated so external consumers get a slim package to import if they only want constants.

Reply:
I'm putting this into my Eth branch and decided to document it, so in that file I have:

```go
// SafeHeightDistance is the distance from the current head that we call "safe" for the purpose of
// assuming finality when relying on EC. It's not a substitute for ChainFinality, but a compromise
// where it's acceptable to have a nonzero margin of error in the finality assumption.
const SafeHeightDistance = 200
```

Feel free to take and edit, I'll rebase ontop of this when it's merged.

Reply:
No definitive consensus on this, which leaves me to pick a more conservative approach and revisit once more data is collected. Leaving this to be 200.

Reply:
Apologies. Due to cached UI i missed your comment. Thanks for cleaning this up in your PR 🙏

Reply:
done in https://github.com/filecoin-project/lotus/pull/13026, copied even more of your test cases for my own versions of them while I was at it

Reply:
thanks